app/controllers/metrics_controller.rb

```diff
@@ -1,19 +1,26 @@
 class MetricsController < ApplicationController
+  include Concerns::DateRangeParser
+
   DEFAULT_METRICS = %w[pageviews unique_pageviews number_of_internal_searches].freeze
 
   def show
     service = MetricsService.new
 
+    date_range = parse_date_range(params[:date_range])
+    from = date_range[:from]
+    to = date_range[:to]
+
     service_params = {
       base_path: params[:base_path],
-      from: params[:from],
-      to: params[:to],
+      from: from,
+      to: to,
       metrics: DEFAULT_METRICS
     }
 
     @summary = SingleContentItemPresenter
       .parse_metrics(service.fetch(service_params))
```

Comment:
I know this is not part of your codebase, but the methods: `parse_metrics`, `parse_time_series`, `set_date_range` doesn't feel like they belong to the `SingleContentItemPresenter`. They feel like internal behaviour that could run after the initialization (after construction phase)

We should be able to rewrite lines 20-23 into:

```ruby
SingleContentItemPresenter.new(base_path: ---, date_range: --- )
```